fixed thread stop

TausifIqbal opened the pull request.

subhash011 commented:
Change the GetLocalIPAddress in ServerCommunicator to get the IP address that does not end with a ".1".

TausifIqbal commented:
changed GetLocalIPAddress , now it returns first IP of machine which dose not  end with ".1"

added SendSocketListenerServer :- It dequeues a packet (server side) , gest destination from serialized data field of packet and writes into the corresponding client socket after forming a string

added SendSocketListenerClient :-  It dequeues a packet (client side) , forms a string and writes into socket

added RecieveSocketListener :- It listens for incoming message both client and server side and after reading the message it forms a packet object and pushes into the queue.
